### DISCUSSION TITLE
CLI tools: expected behaviour when copying system.top for non-OPLS forcefields?

### DISCUSSION THREAD

**Discussion by a-ws-m** [Becksteinlab/MDPOW#265] [Q&A] [answered]
Hi, I've been trying to use MDPOW to compute partition coefficients with the CHARMM forcefield. The first thing I've noticed is that the `system.top` file that is copied by `mdpow-equilibrium` (from [here](https://github.com/Becksteinlab/MDPOW/blob/develop/mdpow/templates/system.top)) has the `oplsaa.ff` directory and files hardcoded. Is this intended behaviour?

**Comment by a-ws-m** [comment]
I'm drafting a PR that improves the way non-OPLS force fields are handled and I think I've discovered that this isn't expected behaviour, see [here](https://github.com/Becksteinlab/MDPOW/blob/3971e5d3c90e2d9e440429b4bcce4513f1c037f8/mdpow/equil.py#L319). I'll include a fix for this in the PR.

**Reply by orbeckst** [reply]
Ok, thanks!

I am not checking this repo regularly so please feel free to ping me on a PR with @orbeckst .

**Comment by orbeckst** [comment]
@VOD555 you've been doing simulations with different FF. 

Have we documentation on how we do this normally?

**Reply by VOD555** [reply]
This [line](https://github.com/Becksteinlab/MDPOW/blob/3971e5d3c90e2d9e440429b4bcce4513f1c037f8/mdpow/templates/system.top#L9) will replace the path in the template topology file with the force field directory defined at [here](https://github.com/Becksteinlab/MDPOW/blob/3971e5d3c90e2d9e440429b4bcce4513f1c037f8/mdpow/forcefields.py#L276C4-L276C4) using the keyword `forcefield`. 

Currently, we only support 3 specific force fields (they are in the documentation ). If one wants to use other force fields, it's better to run the equilibrium by himself/herself and then use the `mdpow-fep` or `fep` function to import the output of the equilibrium simulation and set the fep simulations.

**Comment by a-ws-m** [comment, marked as answer]
Actually I'm mistaken, sorry! I hadn't realised there was a `forcefield` option for the YML file, as it doesn't show up in the `benzene.yml` example. I'm still making that PR though as I've got some changes that should make using arbitrary forcefields easier.

**Reply by orbeckst** [reply]
Our documentation does need some work... I typically look at [templates/runinput.yml](https://github.com/Becksteinlab/MDPOW/blob/develop/mdpow/templates/runinput.yml) to see what's there but I also didn't remember that there's a `forcefield:` key.

Do you want to mark your own comment above as an answer?
